### ISSUE TITLE
Simpler values for the headings font size and line height

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#8069]
Splitting this out from #7969.

I'd like to propose to reconsider the font-size values and consequently the line height values. Ideally, the font-size computed values should translate to values in pixels without decimals. Browsers have different font rendering engines and different rounding algorithms. The only way to avoid inconsistencies is to avoid rounding.

For clarity, I'm not proposing to use pixel values. Font sizes in `em` are fine, but the em values should produce a computed value in pixels without decimals.

I guess the current values for the headings have been set with some typographic hierarchy criteria but I don't understand the math behind them. For example, why H1 is 2.44em? Or H2 1.95em, etc.? Where's the math?

Based on the font size values, the line-heights can be calculated with the same principle, avoiding computed values with decimals.

Here's how the various values are translated in pixels:
```
h1
2.44em x 16 = 39.04px
39.04 x 1.2 = 54.656px

h2
1.95em x 16 = 31.2px
31.2 x 1.4 = 43.68px

h3
1.56em x 16 = 24.96px
24.96 x 1.4 = 34.944px

h4
1.25em x 16 = 20px
20 x 1.5 = 30px

h5
1em x 16 = 16px
16 x 1.8 = 28.8px

h6
0.8em x 16 = 12.8px
12.8 x 1.8 = 23.04px
```

For example, instead of 
```
h1
2.44em x 16 = 39.04px
```

why not to use a simpler value which gets computed to a value in pixels without decimals? This way, setting a proper value for the line height too would be way simpler and would reduce inconsistency across browsers. For example:
```
h1
2.5em x 16 = 40px
```

I'd also suggest to produce computed pixel values with even numbers, for example `40` instead of `39` and do the same for the line-heights. In my experience, if one of the numbers is even and the other is odd, then browsers may place the font within the line-height in an inconsistent way. Say we have a font of 40 pixels and a line-height of 55 pixels. The remainder is 15. Browsers can't place the font in the middle of the line. They will use ascender and descender spaces of 8 and 7 pixels but there's no guarantee they will do it in the same way. And sub-pixel rendering has still a long way to go.

FWIW, I'd like to see better values also for normal text in Gutenberg and in core 😉where the default values are as follows:
```
body
font-size: 13px;
line-height: 1.4em; // 18.2 pixels

paragraphs:
font-size: 13px;
line-height: 1.5; // 19.5 pixels
```

**Comment by aristath** [comment]
> I guess the current values for the headings have been set with some typographic hierarchy criteria but I don't understand the math behind them. For example, why H1 is 2.44em? Or H2 1.95em, etc.? Where's the math?

Just did a quick calculation and it looks like they're using a Major Third [type-scale](https://type-scale.com/) which is 1.25
Perhaps the type-scale ratio should be configurable? 1.25 is not always the best option, and it would be a lot easier to just say `typeScale: 1.33` instead of having to define font-sizes for each and every header-size

**Comment by afercia** [comment]
Thanks @aristath 🙂I see the benefit of using an established typographic scale and I'm not opposed to that. However, we shouldn't live in the illusion browsers will render the scale exactly as it's set. For at least two reasons:
- browsers round in different ways
- a value, for example, of 31.2 pixels works in terms of CSS pixels but on a device screen what `0.2` pixels is? On a screen, 1 pixel is the smallest size unit. So it makes sense to me to round some of the scale numbers to reduce inconsistencies.

**Comment by maddisondesigns** [comment]
Just as an fyi, it's recommended that `line-height` should be a number (i.e. multiplier), rather than specifying a unit such as px or em. e.g. `line-height: 1.2;`

**Comment by afercia** [comment]
Yes, line-height should be "unitless" (unless there's a specific need to use a unit, e.g. vertical alignment etc.). Gutenberg does unitless for the most part; core not so much, unfortunately.

**Comment by aristath** [comment]
I'm not saying we should leave rounding up to browsers... But given a scale number I'm sure we can calculate and round the font-sizes. What I was suggesting is just a way to define all font-sizes given just 2 values. With a base font-size and a scale number we can calculate the rest quite easily and round them to integer values

**Comment by afercia** [comment]
Just to confirm browsers round in different ways, see in the two screenshots below the rendered line-height of the paragraph is different in Chrome and Firefox. There's a subtle difference of 1 pixel. However, if you measure the total height of the paragraph, the difference is more noticeable (196 vs. 201):

<img width="819" alt="screen shot 2018-07-23 at 11 55 40" src="https://user-images.githubusercontent.com/1682452/43070504-4966f5e4-8e70-11e8-91d5-6b1890a246f8.png">

<img width="817" alt="screen shot 2018-07-23 at 11 55 23" src="https://user-images.githubusercontent.com/1682452/43070503-494abe10-8e70-11e8-87ea-716a4e82782d.png">

**Comment by afercia** [comment]
Technically, #8373 is a duplicate 😉 I have no objections though, as long as the details mentioned here don't get lost.
